### ISSUE TITLE
For Linux, appimage, Standard Note update simply does not work

### ISSUE THREAD

**Issue by WritingSolutions2020** [standardnotes/forum#347]
Linux Mint, MATE desktop, 32-bit. But not sure the Linux distribution is the issue here...

**EDIT:** Unless one truly understands Linux, and so can use root privileges to perform a command-line update, updating Standard Notes on Linux simply does not work.

1. There's NO way to update from the GUI. One _must_ use the command line interface (CLI).

2. Using `sudo` to start SN from the command line does NOT enable one to perform an update from the GUI. The command line indicates activity but the update does NOT occur.

3. One must manually locate and find the updated appimage, download it, get it in the desired folder with the correct owner and privileges, then start it with `sudo` privileges to get it to install.

What a pain in the sitzon!

This is truly annoying. What's needed is:

1. GUI feedback, or at least _some_ sensible feedback, about what is and is not happening.

2. The update process puts the updated appimage in the same folder as the version that's being used to update SN.

This has become especially important given SN's seemingly once-per-week update needs. I "get" that updates can be beneficial, but on Linux it becomes an issue of stopping work and performing the functional equivalent of a system administration task rather than simply updating and application...

Thanks,
Riley
SFO

**Comment by mobitar** [comment]
We could add more GUI around this. Just a matter of improving over time. Recently we added the Updates menu which really helped provide more clarity (much more opaque before that). The only thing missing now would be some sort of progress indicator. But yeah, I imagine we'll do stuff here. Just other things with higher priority to be done first.

**Comment by WritingSolutions2020** [comment]
See my rewrite of the issue. At this point it seems _impossible_ to update the SN appimage on Linux from the GUI. Sigh...

**Comment by mobitar** [comment]
@WritingSolutions2020 still unable to replicate your issue, and no one else seems to have the same problem. Might be because you're 32 bit. Need to investigate more when I get the chance. But, this is the reason we added the Updates menu with "Manual download". In case auto doesn't work. This way at least you know there's a new version.

**Comment by WritingSolutions2020** [comment]
Part of the problem may be that I've put the executable in `/usr/local/StandardNotes` and made its owner `root`. I've done that to align SN's installation directory with similar programs such as Postman. With this in mind, it'd be great if SN would:

1. "Know" where its executable is and update to that folder. (I instead find updated appimage files in a hidden directory in my home folder...)

2. Ask for the necessary permissions via a prompt ala other software updates.

**Comment by mobitar** [comment]
It should definitely know where it's located and update the file in place in your existing directory. But it may be that because it's placed in a folder it doesn't have permissions to, it defaults back to your home directory.

**Comment by WritingSolutions2020** [comment]
Updating on my computer remains a complete pain in the sitzon. I need to updgrade as root. Wouldn't it be great, he said, if the SN upgrade process could ask for the user's SUDO credential and move forward without so much manual hoops-jumping...?

**Comment by WritingSolutions2020** [comment]
AND ALSO, IF the update process's directory hierarchies included a constant-value link to the current version, THEN one wouldn't need to go 'round revising shortcuts whose destination points to a specific version. For example, one updates from V34 to V36. If the update process provided a link to "application" or the like that in turn pointed to the current APPIMAGE, then all shortcuts would remain valid. As it is one must update "34" to "36" in the shortcuts' command line...

**Comment by WritingSolutions2020** [comment]
Check how Postman does it. MUCH easier upgrade AND has a single _Postman_ link that runs the most recently updated version of the program...

**Comment by mobitar** [comment]
Hey Riley, definitely hear you that it can be frustrating. The reason it's not so trivial to just go in and fix all these issues and make it absolutely perfect is that we don't control any of the update process. It's all built by this huge framework known as [electron-builder](https://github.com/electron-userland/electron-builder). We're simply at their mercy for how autoupdate is handled. It does a good job, generally speaking, but it's really hard to customize precisely for our app, given it's built generically. It's also a really popular framework, so the developer can't always keep up with the incoming issues.

In any case, the best we can do is continue pulling in new versions of electron-builder as they are released, but that's also risky, since new releases have been known to cause other unrelated issues. So I update it slowly.

Ideally, at some point in the unknown future, we can develop our own "simple" style autoupdate functionality, but, I imagine it would be a very large undertaking.

**Comment by WritingSolutions2020** [comment]
Thanks Mo. All I can tell you is that it is the single most difficult and time consuming program to update on my Linux computer. I _dread_ seeing that there's a new version...

**Comment by WritingSolutions2020** [comment]
And after jumping through those hoops, the client informs me that I need to do Synch Resolution. Sigh... Please Lord, make it stop!

**Comment by WritingSolutions2020** [comment]
I am of today giving up on the Linux 32-bit desktop client -- it's just too much of a pain in the ass to deal with. I'll add that as much as I enjoy Standard Notes and appreciate all the work you do, Standard Notes appimage for Linux is in my experience flat out the most demanding, fragile, and frustrating program I run on this system. Seriously.

Upgraded to 3.0.14. Which requires quite a few hand motions. Perhaps because I run SN in /usr/local/standardnotes. Requires me to use sudo to copy the app in, then run around updating shortcuts to point to the version number in the appimage's name. No other program is this difficult to upgrade...

Encountered multiple sync issues. Tried using Batch Manager to remove duplicate / sync-conflicted notes. It only partially worked. So I had to (1) delete the duplicates, (2) go over trash and clean them up there, too. Sigh...

Next, pretty much all extensions had two instances. Batch Manager didn't help with those so had to click-click-click-click-click delete the duplicates.

So it's the web client for me from here on out. Perhaps when I upgrade my tired old 32-bit laptop I'll try the desktop client again.

Cheers,
Riley
SFO
